### PR TITLE
Feature/remove filters

### PR DESCRIPTION
**Pull request by AD9018**
Fixed long-awaited issue that dealt with x'ing out filter chips in cases table and reflecting changes in criteria. Tested in multiple ways to try to throw an error but seems consistent in all cases. The reason I tackled this was that it was one of the Trello cards that would significantly boost the user experience using functional components. 

Fixes # (issue)
1. Users can now X out filter chips with ease and not go back into the ANT component and reset. The biggest issue and challenge was abstracting the ANT component and the search and filter functionality because they are separate. Everything inside the table is from ANT design. We have the chips being able to be x'd out outside of the ANT table component which was done using the document object model otherwise it would not have been recognized by Reacts "virtual DOM". We needed to hit the actual DOM to extend changes. 

2. A minor fix that made a big difference which I will include in this pull req. would be depending on what criteria you search it would show up in the filter chip instead of raw JSON from the backend (horrible user experience) Ex: origin_city but now is showing up as Baltimore if you search Baltimore rather than origin_city. 

## Urgent Front-end change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Comment by AD9018** [comment]
Please see the video for this pull below 

https://www.loom.com/share/a8073174ab2f42b68aeff51e2e28e424

**Review by Fabian-Gorka — APPROVED** [review]
The functionality looks good and the code is very clean and easily read nice job on fixing this problem

**Comment by willmoon19** [comment]
This looks great! I think the style was done well, and you cleanly solved a problem and provided more functionality.

**Comment by matthewblackbu** [comment]
As a Data Scientist this looks very good way to fix this problem not fully understanding the front end it looks like you did a great job editing the code to make it work

**Review by willmoon19 — APPROVED** [review]
This looks great! I think you did a wonderful job solving a major problem and made clean code for adding more functionality

File: src/components/pages/Cases/CaseTable.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ export default function CaseTable(props) {
   };
 
   const { caseData, userInfo, savedCases, setSavedCases } = props;
-
+  // console.log('CASE DATA', caseData)
```
Comment (jenna-anderson) [inline src/components/pages/Cases/CaseTable.js:41]: Could this commented out console log be removed?

File: src/components/pages/Cases/CaseTable.js [hunk 2]
```diff
@@ -178,16 +181,20 @@ export default function CaseTable(props) {
 
   // returns processed array of filters
   const processFilters = filters => {
+    // console.log('FILTER', filters)
```
Comment (jenna-anderson) [inline src/components/pages/Cases/CaseTable.js:184]: Another potential unnecessary commented out console log

File: src/components/pages/Cases/CaseTable.js [hunk 3]
```diff
@@ -215,11 +222,12 @@ export default function CaseTable(props) {
         }
       }
     }
+    // console.log('RESSS', res)
```
Comment (jenna-anderson) [inline src/components/pages/Cases/CaseTable.js:225]: Unneeded commented out console log

File: src/components/pages/Cases/CaseTable.js [hunk 4]
```diff
@@ -390,7 +398,7 @@ export default function CaseTable(props) {
       bookmarks.forEach(b => {
         if (savedIds.includes(b.case_id)) {
           // This should be an alert
-          console.log('Case already saved to bookmarks');
+          // console.log('Case already saved to bookmarks');
```
Comment (jenna-anderson) [inline src/components/pages/Cases/CaseTable.js:401]: Is this one needed? Was it being used as an alert to the user before?

File: src/components/pages/Cases/CaseTable.js [hunk 5]
```diff
@@ -570,11 +596,31 @@ export default function CaseTable(props) {
...
+                    >
+                      X
+                    </span>
+                  </Tag>
+                );
+              })}
             </div>
+
             <div className="case-table">
+              {/* {console.log(rowSelection)} */}
```
Comment (jenna-anderson) [inline src/components/pages/Cases/CaseTable.js:623]: Do we need this console log?

File: src/components/pages/Cases/CaseTable.js [hunk 6]
```diff
@@ -549,6 +557,24 @@ export default function CaseTable(props) {
...
+    filterState[keyWord] = null;
+
+    setInitialFilters(filterState);
+
+    document.getElementById(keyWord).click();
+
+    return;
+  };
+
+  // console.log('INITIAL FILTERS', initialFilters)
```
Comment (joshuascan) [inline src/components/pages/Cases/CaseTable.js:577]: Would be good to remove this console log if no longer needed

**Review by jenna-anderson — COMMENTED** [review]
Excellent work. Your description and video were helpful in understanding the changes you made. There are some commented out console logs that could be removed - see inline comments

File: src/components/pages/Cases/CaseTable.js [hunk 7]
```diff
@@ -570,11 +596,31 @@ export default function CaseTable(props) {
           <TabPane tab="Initial Cases" key="1">
             <div>
               Filters:{' '}
-              {processFilters(initialFilters).map(filter => (
-                <Tag key={filter}>{filter}</Tag>
-              ))}
+              {processFilters(initialFilters).map(filter => {
+                console.log('EACH FILTER ', filter);
```
Comment (joshuascan) [inline src/components/pages/Cases/CaseTable.js:600]: Same for this one and the one on line 623.

File: src/components/pages/Cases/CaseTable.js (hunk 6 above)
Comment (joshuascan) [inline src/components/pages/Cases/CaseTable.js:577]: Might be good to remove this console log if no longer needed

**Review by joshuascan — APPROVED** [review]
Good job on this. It's a good solution for what seems to have been a very stubborn problem. I would maybe just remove the console logs as noted. Nice work!

File: src/components/pages/Cases/CaseTable.js [hunk 8]
```diff
@@ -555,20 +563,18 @@ export default function CaseTable(props) {
...
+
+  const removeSearchTerm = (filterState, keyWord) => {
+    filterState[keyWord] = null;
+
+    setInitialFilters(filterState);
+
+    document.getElementById(keyWord).click();
+
+    return;
   };
```
Comment (chrisjs87) [inline src/components/pages/Cases/CaseTable.js:576]: I might say the spacing for this function is different/not standardized with others throughout this file. You may want to remove the extra lines of spacing in 569, 571, and 573. Consistency is useful, but entirely a nitpick.

**Review by chrisjs87 — APPROVED** [review]
The functionality of this code is excellent. I noticed one discrepancy wherein spacing could be more standardized to reflect the rest of the document. See comment therein, line 576.

**Review by ZacharyCooremans — APPROVED** [review]
Looks really good. Everyone made good comments and I don't see anything else to add. Way to get this feature working and making a great PR

File: src/components/pages/Cases/CaseTable.js (hunk 1 above)
Comment (paulstgermain) [reply src/components/pages/Cases/CaseTable.js:41]: Agreed - this should be removed before this PR gets Merged!

File: src/components/pages/Cases/CaseTable.js (hunk 4 above)
Comment (paulstgermain) [reply src/components/pages/Cases/CaseTable.js:401]: Let's pin this one for another time - we may want to change this to an alert of some kind before we ship this app.

**Review by paulstgermain — CHANGES_REQUESTED** [review]
To start, I'd love to leave some feedback on the impressively thorough walkthrough Loom you left attached to this PR:

- The flow of your explanation was great, by first showing the Trello board, then an example of the front-end as it used to appear thus highlighting the problem, then the front-end as it looks now, before diving into the code that makes this solution work, you lead your listener through the solution in a very cohesive, orienting manner.

- You clearly noted the issue you sought to fix by informing your listener that your front-end was rendering the raw JSON data which wasn't a great user experience.

- You did nicely flexing your technical expertise and ability to quickly learn new frameworks by explaining that the filter and filter chips are in fact NOT Ant Design and that tapping the actual DOM was the solution to a problem that we'd been struggling with for well over a month.

The only critiques I have for you at this moment involve confidence when speaking about your work - this is something we **all** need practice with, to be clear! But I feel it's important to note that it's easy to get mentally caught up with whether we're explaining things well enough or not, and as a result, we lose our train of thought. There were a couple of spots where this seemed to almost be the case, where you could have pushed your excellent explanation just a step further by recalling the technical term you were about to drop, and I'd love to see you approach these instances even more confidently in the future!

Regarding the code itself, you produced an elegant solution to a problem the team has been facing for about a month now. By tapping the DOM, you were able to effectively change state with a lick handler every time one of the 'x' buttons was pressed by realizing that the filter feature was, in fact, not a part of ANT Design - the end result being a fully working feature that improves the user experience greatly.

As a final note though, I would love to see the `console.log`s that still have unresolved conversations connected to them removed, and those conversations resolved before I give that final approval, but otherwise very well done!

**Review by paulstgermain — APPROVED** [review]

**Pull request merged**